domain/storage/state/storagepool.go

```diff
@@ -181,32 +224,52 @@ func (st State) DeleteStoragePool(ctx context.Context, name string) error {
 		return errors.Capture(err)
 	}
 
-	poolAttributeDeleteQ := `
-DELETE FROM storage_pool_attribute
-WHERE  storage_pool_attribute.storage_pool_uuid = (select uuid FROM storage_pool WHERE name = $M.name)
-`
+	pool := StoragePool{Name: name}
 
-	poolDeleteQ := `
+	poolAttributeDeleteStmt, err := st.Prepare(`
+DELETE FROM storage_pool_attribute
+WHERE  storage_pool_attribute.storage_pool_uuid = (
+  select uuid FROM storage_pool WHERE name = $StoragePool.name
+)`, pool)
+	if err != nil {
+		return errors.Capture(err)
+	}
+	poolDeleteStmt, err := st.Prepare(`
 DELETE FROM storage_pool
-WHERE  storage_pool.uuid = (select uuid FROM storage_pool WHERE name = $M.name)
-`
-
-	poolAttributeDeleteStmt, err := st.Prepare(poolAttributeDeleteQ, sqlair.M{})
+WHERE  storage_pool.uuid = (
+  select uuid FROM storage_pool WHERE name = $StoragePool.name
+)`, pool)
 	if err != nil {
 		return errors.Capture(err)
 	}
-	poolDeleteStmt, err := st.Prepare(poolDeleteQ, sqlair.M{})
+
+	originStmt, err := st.Prepare(`
+SELECT (spo.*) AS (&poolOrigin.*)
+FROM   storage_pool sp
+       LEFT JOIN storage_pool_origin spo ON spo.id = sp.origin_id
+WHERE  sp.name = $StoragePool.name`, pool, poolOrigin{})
 	if err != nil {
 		return errors.Capture(err)
 	}
 
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		nameMap := sqlair.M{"name": name}
-		if err := tx.Query(ctx, poolAttributeDeleteStmt, nameMap).Run(); err != nil {
+		var origin poolOrigin
+		err := tx.Query(ctx, originStmt, pool).Get(&origin)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Errorf("storage pool %q not found", name).Add(storageerrors.PoolNotFoundError)
+		}
+		if err != nil {
+			return errors.Errorf("getting storage pool origin: %w", err)
+		}
+		if origin.Origin != string(domainstorage.StoragePoolOriginUser) {
```

Comment:
Can I suggest that we do this check in the service layer before the state layer? We can then dust update the delete sql where clause to only consider origin id of user.

This is fine because we are never going to allow the origin of a pool to change.

Reply:
The `trg_storage_pool_immutable_delete` trigger now enforces this.